test/integration_test/common_test.go

```diff
@@ -135,21 +116,69 @@ func constructStorageCluster() (*corev1.StorageCluster, error) {
 }
 
 func createStorageClusterFromSpec(filename string) (*corev1.StorageCluster, error) {
-	filepath := path.Join(specDir, filename)
+	filepath := path.Join(testutil.SpecDir, filename)
 	scheme := runtime.NewScheme()
 	cluster := &corev1.StorageCluster{}
 	if err := k8sutil.ParseObjectFromFile(filepath, scheme, cluster); err != nil {
 		return nil, err
 	}
-	if err := createStorageCluster(cluster); err != nil {
+	cluster, err := createStorageCluster(cluster)
+	if err != nil {
 		return nil, err
 	}
 	return cluster, nil
 }
 
-func createStorageCluster(cluster *corev1.StorageCluster) error {
-	_, err := operator.Instance().CreateStorageCluster(cluster)
-	return err
+func createStorageCluster(cluster *corev1.StorageCluster) (*corev1.StorageCluster, error) {
+	cluster, err := operator.Instance().CreateStorageCluster(cluster)
+	return cluster, err
+}
+
+func updateStorageCluster(cluster *corev1.StorageCluster, specGenURL string, imageListMap map[string]string) (*corev1.StorageCluster, error) {
+	// Get StorageCluster
+	cluster, err := operator.Instance().GetStorageCluster(cluster.Name, cluster.Namespace)
+
+	// Set Portworx Image
+	cluster.Spec.Image = imageListMap["version"]
+
+	// Set release manifest URL in case of edge-install.portworx.com
+	if strings.Contains(specGenURL, "edge") {
+		releaseManifestURL, err := testutil.ConstructPxReleaseManifestURL(specGenURL)
+		if err != nil {
+			return nil, err
+		}
+
+		cluster.Spec.CommonConfig = corev1.CommonConfig{
+			Env: []v1.EnvVar{
+				{
+					Name:  testutil.PxReleaseManifestURLEnvVarName,
+					Value: releaseManifestURL,
+				},
+			},
+		}
+
+		// Add Portwrox Docker Credentials
+		if pxDockerUsername != "" && pxDockerPassword != "" {
+			newEnvVar := []v1.EnvVar{
+				{
+					Name:  testutil.PxRegistryUserEnvVarName,
+					Value: pxDockerUsername,
+				},
+				{
+					Name:  testutil.PxRegistryPasswordEnvVarName,
+					Value: pxDockerPassword,
+				},
+			}
+			cluster.Spec.CommonConfig.Env = append(cluster.Spec.CommonConfig.Env, newEnvVar...)
+		}
+	}
```

Comment:
Can we extract these lines into a method that can be used by constructStorageCluster as well?

Reply:
Created a func out of code block.